runtests.py

```diff
@@ -130,11 +130,12 @@ def main(argv):
     args = parser.parse_args(argv)
 
     if args.pep8:
-        # os.system("flake8 scipy --ignore=F403,F841,F401,F811,F405,E121,E122,"
-        #           "E123,E125,E126,E127,E128,E226,E231,E251,E265,E266,E302,"
-        #           "E402,E501,E712,E721,E731,E741,W291,W293,W391,W503,W504"
-        #           "--exclude=scipy/_lib/six.py")
+        # Lint the source using the configuration in tox.ini.
         os.system("pycodestyle scipy benchmarks/benchmarks")
```

Comment:
Shouldn't this be also a `subprocess.run` call?

Reply:
@ilayn 
RE: 
`         os.system("pycodestyle scipy benchmarks/benchmarks")`

I'm not even sure we need that line?
On a local branch checkout, I commented out the line, and still caught things. Is this line catching the 
top level python files only?  (e.g. `runtests.py`, `setup.py`, etc) seems weird to put something for that separate from everything else. 

Incidentally, removing the line it speeds up execution substantially.

I then removed the W291 ignore from the ini file (on local PR branch) and added a trailing whitespace (probably my most frequent pep8 violation). The error is still caught without the line. 

```
(scipy-dev) Lucass-MacBook:scipy rlucas$ git diff --staged
diff --git a/runtests.py b/runtests.py
index 4f33707dc..16799add2 100755
--- a/runtests.py
+++ b/runtests.py
@@ -65,8 +65,6 @@ except ImportError:  # old Python
     from imp import new_module
 
 ROOT_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__)))
-
-
 def main(argv):
     parser = ArgumentParser(usage=__doc__.lstrip())
     parser.add_argument("--verbose", "-v", action="count", default=1,
@@ -130,7 +128,7 @@ def main(argv):
 
     if args.pep8:
         # Lint the source using the configuration in tox.ini.
-        os.system("pycodestyle scipy benchmarks/benchmarks")
+        # os.system("pycodestyle --show-source --show-pep8 scipy benchmarks/benchmarks") 
         # Lint just the diff since branching off of master using a
         # stricter configuration.
         lint_diff = os.path.join(ROOT_DIR, 'tools', 'lint_diff.py')
@@ -475,3 +473,4 @@ def lcov_generate():
 
 if __name__ == "__main__":
     main(argv=sys.argv[1:])
+
diff --git a/tools/lint_diff.ini b/tools/lint_diff.ini
index 3cf2106e5..117c0d6eb 100644
--- a/tools/lint_diff.ini
+++ b/tools/lint_diff.ini
@@ -1,5 +1,5 @@
 [pycodestyle]
 max_line_length = 79
 statistics = True
-ignore = E121,E122,E123,E125,E126,E127,E128,E226,E251,E265,E266,E302,E402,E501,E712,E721,E731,E741,W291,W293,W391,W503,W504
+ignore = E121,E122,E123,E125,E126,E127,E128,E226,E251,E265,E266,E302,E402,E501,E712,E721,E731,E741,W293,W391,W503,W504
 exclude = scipy/__config__.py
(scipy-dev) Lucass-MacBook:scipy rlucas$ python runtests.py --pep8
./runtests.py:131:89: W291 trailing whitespace
1       W291 trailing whitespace
```

Unless there is a reason to keep it I'd suggest we remove the line, timings:
```
(scipy-dev) Lucass-MacBook:scipy rlucas$ time python runtests.py --pep8
./runtests.py:131:89: W291 trailing whitespace
1       W291 trailing whitespace


real	0m0.414s
user	0m0.325s
sys	0m0.086s 
```
now keeping that line in:
```
(scipy-dev) Lucass-MacBook:scipy rlucas$ time python runtests.py --pep8
./runtests.py:131:61: W291 trailing whitespace
1       W291 trailing whitespace


real	0m39.582s
user	0m38.564s
sys	0m0.340s
```

Reply:
The first call is diffing the entire source tree, and the second call is just linting the diff. Perhaps the former is redundant now, but I can't swear to it.

Reply:
@person142 not sure I follow, I don't think:
`os.system("pycodestyle scipy benchmarks/benchmarks")`
does any diff, unless that is what `pycodestyle` does under the hood but that should be opaque to the caller. 

My understanding is that the `tools/lint.py` is the script where the calls are being made to the diffs. 

Did I misunderstand what you meant?

The call:
`os.system("pycodestyle scipy benchmarks/benchmarks")`
 doesn't seem to capture and forward to `stdout` so if you wanted that then it might make sense to switch but it does seem like if that is what you want to do that should reside in the linter file and not in the test harness but that is just my 2 cents.